### PR TITLE
[explorer][delegation] add wallet connection to nav header

### PR DESCRIPTION
**Pull request by jillxuu**
* very rough version of wallet connection on explorer header nav bar. will make UI rich in future iterations.

https://user-images.githubusercontent.com/121921928/212138507-cf8c1024-b226-4fc2-ad60-d60d3864934e.mov

### CODE REVIEW

**Comment by 0xZihan** [comment]
@0xmaayan Our dev mode is not exactly a secret. Anyone who knows the link can access it. I think the answer is no but want to double check with you, is there any safety concern that we enable wallet connection now? I guess it's not exactly limited to dev mode. Is there any safety concern to enable wallet connection on explorer in general?

**Comment by 0xmaayan** [comment]
> @0xmaayan Our dev mode is not exactly a secret. Anyone who knows the link can access it. I think the answer is no but want to double check with you, is there any safety concern that we enable wallet connection now? I guess it's not exactly limited to dev mode. Is there any safety concern to enable wallet connection on explorer in general?

I dont think there is any security concern other than the regular concerns lol

One thing you can do for when in dev mode (or even after, depends on the product) is to "disable" the site when wallet is on a specific network. You can take a look on how we did it with ANS (try go to aptosnames.com and have your wallet on testnet).
We have a plan to add this functionality on the adapter itself.

File: src/pages/layout/ConnectWallet.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    setCopiedTooltipOpen(true);
+
+    setTimeout(() => {
+      setCopiedTooltipOpen(false);
+    }, COPIED_TOOLTIP);
+  };
+
+  return (
+    <>
+      {connected ? (
```
Comment (0xmaayan) [inline src/pages/layout/ConnectWallet.tsx:43]: Wallet adapter has a UI package that can help with all of that

https://github.com/aptos-labs/aptos-wallet-adapter/tree/main/packages/wallet-adapter-ant-design

**Review by 0xmaayan — APPROVED** [review]
adapter implementation looks good. 
I think we can use the wallet connect/selector design and have it as a standard UI that can be used in the adapter UI package - maybe even built it as a package in the adapter and use it here

File: src/index.tsx [hunk 2]
```diff
@@ -41,14 +48,25 @@ declare global {
...
+  new PetraWallet(),
+  new PontemWallet(),
+  new MartianWallet(),
+  new FewchaWallet(),
+  new RiseWallet(),
+  new SpikaWallet(),
+];
+
 // delay rendering the application until the window.onload event has fired when integrating with the window.aptos API
 window.addEventListener("load", () => {
```
[the remark this answers is not in the document]
Comment (jillxuu) [reply src/index.tsx:61]: would you mind clarifying here? why once the change to use new adapter's landed, this event listener can be removed?

[the remark this answers is not in the document]
Comment (0xmaayan) [reply src/index.tsx:61]: I think this event was introduced when Governance was part of the Explorer, and we only used Petra wallet. Due to some race condition with the wallet injection and the browser window we had to add this event. 
Now that we use the wallet adapter, we can remove as the adapter handles it.
cc @0xZihan

**Pull request merged**